### ISSUE TITLE
[BUG] genai.evaluate fails with "Yaml file exists" on second run if global random seed is set

### ISSUE THREAD

**Issue by JonahMelchers** [mlflow/mlflow#18844]
### Issues Policy acknowledgement

- [x] I have read and agree to submit bug reports in accordance with the [issues policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)

### Where did you encounter this bug?

Local machine

### MLflow version

- Client: 3.5.1

### System information

- **OS Platform**: macOS 15.7
- **Python version**: 3.11.13

**python packages**:
- mlflow==3.5.1
- pandas==2.3.2
- accelerate==1.10.1
- torch==2.8.0
- transformers==4.56.2

[requirements.txt](https://github.com/user-attachments/files/23544779/requirements.txt)

### Describe the problem

When running an MLflow experiment that involves both transformers.Trainer and mlflow.genai.evaluate, a conflict arises in the MLflow tracing mechanism.

The transformers.Trainer initialization process calls a utility function (set_seed) that sets a global, fixed seed for Python's random module (e.g., random.seed(42)).

If mlflow.genai.evaluate is called later in the same process, it appears to use this global random state to generate trace IDs. This leads to the following behavior:

The first run within an MLflow experiment succeeds without any issues.
All subsequent runs of the same script within the same experiment fail during the mlflow.genai.evaluate call, producing repeated warnings and causing the evaluation not log results.
The specific warning message is:

> WARNING mlflow.tracing.export.mlflow_v3: Failed to send trace to MLflow backend: Yaml file '/path/to/mlruns/.../traces/tr-.../trace_info.yaml' exists as '/path/to/mlruns/.../traces/tr-.../trace_info.yaml'

This suggests that because the random seed is fixed, genai.evaluate attempts to generate the same trace IDs on the second run, leading to a file collision in the MLflow artifacts directory.

The workaround is to run the random.seed() function before calling mlflow.genai.evaluate() to reset the generator and prevent trace ID collisions.

### Tracking information

```shell
REPLACE_ME
```

### Code to reproduce issue

```
import mlflow
from mlflow.genai import scorer, evaluate
from mlflow.tracking import MlflowClient
import pandas as pd
from transformers import AutoModelForCausalLM, TrainingArguments, Trainer
from transformers.trainer_utils import set_seed


def set_experiment(num: int):
    experiment_name = f"Trace_ID_Collision_Demo_v{num}"     # should be a new one
    client = MlflowClient()
    experiment = client.get_experiment_by_name(experiment_name)
    if experiment is not None:
        raise Exception(f"Experiment with name {experiment_name} already exists!")
    mlflow.set_experiment(experiment_name)

def run_experiment():
    with mlflow.start_run():

        if True:
            tmp_dir = "./trash"
            model = AutoModelForCausalLM.from_pretrained("openai-community/gpt2", cache_dir=tmp_dir)
            args = TrainingArguments(output_dir=tmp_dir)
            Trainer(model=model, args=args)  
            # The problem therefore also occurs with SFTTrainer from trl.
            # Because internally it also uses the transformers Trainer.
        else:
            # This function is the reason for everything.
            set_seed(42)

        data = pd.DataFrame(
            [
                {
                    "inputs": {"question": "What is MLflow?"},
                    "outputs": "MLflow is an ML platform",
                    "expectations": {'expected_response': "MLflow is an ML platform"},
                },
                {
                    "inputs": {"question": "What is Spark?"},
                    "outputs": "I don't know",
                    "expectations": {'expected_response': "Spark is a data engine"},
                },
            ]
        )

        @scorer
        def exact_match(expectations: dict[str, str], outputs: str):
            return expectations['expected_response'] == outputs
        
        # Workaround
        # import random
        # random.seed()

        evaluate(
            data=data,
            scorers=[exact_match]
        )

if __name__ == "__main__":
    set_experiment(0)

    print("▶️  Running experiment for the first time (should succeed)...")
    run_experiment()

    print("▶️  Running experiment for the second time (will show warnings and fail to send trace)...")
    run_experiment()

```

### Stack trace

```
2025/11/14 11:06:21 INFO mlflow.tracking.fluent: Experiment with name 'Trace_ID_Collision_Demo_v0' does not exist. Creating a new experiment.
▶️  Running experiment for the first time (should succeed)...
Evaluating: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2/2 [Elapsed: 00:00, Remaining: 00:00] 

✨ Evaluation completed.

Metrics and evaluation results are logged to the MLflow run:
  Run name: nosy-fly-764
  Run ID: 8ebf2ec853124b90bf1f28dad9a66898

To view the detailed evaluation results with sample-wise scores,
open the Traces tab in the Run page in the MLflow UI.

▶️  Running experiment for the second time (will show warnings and fail to send trace)...
Evaluating:   0%|                                                                                                                            | 0/2 [Elapsed: 00:00, Remaining: ?] 
2025/11/14 11:06:23 WARNING mlflow.tracing.export.mlflow_v3: Failed to send trace to MLflow backend: Yaml file '.../mlruns/513591575296842354/traces/tr-bdd640fb06671ad11c80317fa3b1799d/trace_info.yaml' exists as '.../mlruns/513591575296842354/traces/tr-bdd640fb06671ad11c80317fa3b1799d/trace_info.yaml
2025/11/14 11:06:23 WARNING mlflow.tracing.export.mlflow_v3: Failed to send trace to MLflow backend: Yaml file '.../mlruns/513591575296842354/traces/tr-1a3d1fa7bc8960a923b8c1e9392456de/trace_info.yaml' exists as '.../mlruns/513591575296842354/traces/tr-1a3d1fa7bc8960a923b8c1e9392456de/trace_info.yaml
Evaluating: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2/2 [Elapsed: 00:00, Remaining: 00:00] 

✨ Evaluation completed.

Metrics and evaluation results are logged to the MLflow run:
  Run name: traveling-calf-604
  Run ID: a697fa39d09c49479b41da9a4a36dc30

To view the detailed evaluation results with sample-wise scores,
open the Traces tab in the Run page in the MLflow UI.

```

### Other info / logs

```
REPLACE_ME
```

### What component(s) does this bug affect?

- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/scoring`: MLflow model serving, deployment tools, Spark UDFs
- [x] `area/evaluation`: MLflow model evaluation features, evaluation metrics, and evaluation workflows
- [ ] `area/prompt`: MLflow prompt engineering features, prompt templates, and prompt management
- [x] `area/tracing`: MLflow Tracing features, tracing APIs, and LLM tracing functionality
- [ ] `area/gateway`: MLflow AI Gateway client APIs, server, and third-party integrations
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/uiux`: Front-end, user experience, plotting
- [ ] `area/docs`: MLflow documentation pages

**Comment by soyun11** [comment]
Hi, I'd like to work on a fix for this issue. Please assign me when convenient.

**Comment by soyun11** [comment]
Hi! I’ve submitted PR #18855 with a fix for this issue.
If possible, could you please assign this issue to me and review/merge the PR when you have time?
Thanks a lot!

**Comment by daniellok-db** [comment]
@soyun11 thank you so much for the fix. we'll get it reviewed!
